server/src/main/java/org/elasticsearch/common/settings/LocallyMountedSecrets.java

```diff
@@ -191,20 +205,45 @@ LocalFileSecrets processSecretsFile(Path path) throws IOException {
         try (
             var fis = Files.newInputStream(path);
             var bis = new BufferedInputStream(fis);
-            var parser = JSON.xContent().createParser(XContentParserConfiguration.EMPTY, bis)
+            var parser = JSON.xContent()
+                .createParser(XContentParserConfiguration.EMPTY.withDeprecationHandler(DeprecationHandler.IGNORE_DEPRECATIONS), bis)
         ) {
             return secretsParser.apply(parser, null);
         }
     }
 
-    record LocalFileSecrets(Map<String, String> map, ReservedStateVersion metadata) implements Writeable {
+    record LocalFileSecrets(Map<String, byte[]> map, ReservedStateVersion metadata) implements Writeable {
+
+        public static LocalFileSecrets createLocalFileSecrets(
```

Comment:
nit: the method name is a bit confusing now since it takes file secrets, but that is not the "file secrets" in the method name. Could we simplify the name to just "create" since the class name already has the "local" and "secrets" parts?

Reply:
We don't need the method after moving the logic into the parser.